repositories/pokemon_repo.go

```diff
@@ -48,3 +54,85 @@ func (pr PokemonRepo) WritePokemonCsvFile(pokemons []models.Pokemon) error {
 	}
 	return nil
 }
+
+func (pr PokemonRepo) GetPokemonsFromCSVConcurrently(idType string, items int, itemsPerWorker int) ([]models.Pokemon, error) {
+	numberOfWorkers := items / itemsPerWorker
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+	defer cancel()
+	row := make(chan []string)
+	pokemon := make(chan models.Pokemon)
+	readedItems := 0
+
+	csvfile, err := os.Open(pr.dataFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer csvfile.Close()
+
+	reader := csv.NewReader(csvfile)
+
+	go func() {
```

Comment:
I don't think that this has to go in a go routine, does it?

Reply:
I think it needs to, if I remove the go routine the request never ends 😅  I think because of the connection and close for the row channel